From b349a2915750a5de22339a8bd12ec10ad35d61da Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 17 May 2023 08:13:47 +1000
Subject: [PATCH 1/2] wined3d: rename shader_extract_from_dxbc to
 wined3d_shader_extract_from_dxbc

Stops a linker error of duplicate name when using upstream vk3d3.
---
 dlls/wined3d/shader.c          | 2 +-
 dlls/wined3d/shader_sm4.c      | 2 +-
 dlls/wined3d/wined3d_private.h | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/dlls/wined3d/shader.c b/dlls/wined3d/shader.c
index b4df79508ec..2728929045f 100644
--- a/dlls/wined3d/shader.c
+++ b/dlls/wined3d/shader.c
@@ -2481,7 +2481,7 @@ static HRESULT shader_init(struct wined3d_shader *shader, struct wined3d_device
         shader->byte_code_size = desc->byte_code_size;
 
         max_version = shader_max_version_from_feature_level(device->cs->c.state->feature_level);
-        if (FAILED(hr = shader_extract_from_dxbc(shader, max_version, &source_type)))
+        if (FAILED(hr = wined3d_shader_extract_from_dxbc(shader, max_version, &source_type)))
             goto fail;
 
         if (!(shader->frontend = shader_select_frontend(source_type)))
diff --git a/dlls/wined3d/shader_sm4.c b/dlls/wined3d/shader_sm4.c
index 78abdbe81ac..74391cc1be8 100644
--- a/dlls/wined3d/shader_sm4.c
+++ b/dlls/wined3d/shader_sm4.c
@@ -2077,7 +2077,7 @@ static HRESULT shader_dxbc_process_section(struct wined3d_shader *shader, unsign
     return S_OK;
 }
 
-HRESULT shader_extract_from_dxbc(struct wined3d_shader *shader,
+HRESULT wined3d_shader_extract_from_dxbc(struct wined3d_shader *shader,
         unsigned int max_shader_version, enum vkd3d_shader_source_type *source_type)
 {
     const struct vkd3d_shader_code dxbc = {.code = shader->byte_code, .size = shader->byte_code_size};
diff --git a/dlls/wined3d/wined3d_private.h b/dlls/wined3d/wined3d_private.h
index a586d6847bf..001534e8c38 100644
--- a/dlls/wined3d/wined3d_private.h
+++ b/dlls/wined3d/wined3d_private.h
@@ -1466,7 +1466,7 @@ struct wined3d_shader_frontend
 extern const struct wined3d_shader_frontend sm1_shader_frontend DECLSPEC_HIDDEN;
 extern const struct wined3d_shader_frontend sm4_shader_frontend DECLSPEC_HIDDEN;
 
-HRESULT shader_extract_from_dxbc(struct wined3d_shader *shader,
+HRESULT wined3d_shader_extract_from_dxbc(struct wined3d_shader *shader,
         unsigned int max_shader_version, enum vkd3d_shader_source_type *source_type) DECLSPEC_HIDDEN;
 BOOL shader_get_stream_output_register_info(const struct wined3d_shader *shader,
         const struct wined3d_stream_output_element *so_element, unsigned int *register_idx,
-- 
2.40.1

